cloudflare/resource_cloudflare_logpush_job.go

```diff
@@ -0,0 +1,176 @@
+package cloudflare
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceCloudflareLogpushJob() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudflareLogpushJobCreate,
+		Read:   resourceCloudflareLogpushJobRead,
+		Update: resourceCloudflareLogpushJobUpdate,
+		Delete: resourceCloudflareLogpushJobDelete,
+
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"zone": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"logpull_options": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"destination_conf": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"ownership_challenge": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func getJobFromResource(d *schema.ResourceData) cloudflare.LogpushJob {
+	id, err := strconv.Atoi(d.Id())
+
+	if err != nil {
+		fmt.Errorf("Could not extract Logpush job from resource - invalid identifier: %+v", id)
+	}
+
+	job := cloudflare.LogpushJob{
+		ID:                 id,
+		Enabled:            d.Get("enabled").(bool),
+		Name:               d.Get("name").(string),
+		LogpullOptions:     d.Get("logpull_options").(string),
+		DestinationConf:    d.Get("destination_conf").(string),
+		OwnershipChallenge: d.Get("ownership_challenge").(string),
+	}
+	return job
+}
+
+func resourceCloudflareLogpushJobRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	jobId, err := strconv.Atoi(d.Id())
+
+	zoneName := d.Get("zone").(string)
+	zoneId, err := client.ZoneIDByName(zoneName)
+	if err != nil {
+		return fmt.Errorf("error finding zone %q: %s", zoneName, err)
+	}
+
+	job, err := client.LogpushJob(zoneId, jobId)
+
+	if err != nil {
+		if strings.Contains(err.Error(), "404") {
+			return nil
+		}
+		return fmt.Errorf("error finding logpush job %q: %s", jobId, err)
+	}
+
+	if job.ID == 0 {
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("name", job.Name)
+	d.Set("enabled", job.Enabled)
+	d.Set("logpull_options", job.LogpullOptions)
+	d.Set("destination_conf", job.DestinationConf)
+	d.Set("ownership_challenge", d.Get("ownership_challenge"))
+
+	return nil
+}
+
+func resourceCloudflareLogpushJobCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	job := getJobFromResource(d)
+
+	if d.Get("ownership_challenge") == "" {
```

Comment:
Similar comment to https://github.com/terraform-providers/terraform-provider-cloudflare/pull/287#discussion_r273743243

Reply:
Removed in d7f1163, now that the schema required ownership_challenge.